sdk/communication/Azure.Communication.Common/src/CommunicationTokenRefreshOptions.cs

```diff
@@ -15,31 +15,27 @@ public class CommunicationTokenRefreshOptions
     {
         internal bool RefreshProactively { get; }
         internal Func<CancellationToken, string> TokenRefresher { get; }
-        internal Func<CancellationToken, ValueTask<string>> AsyncTokenRefresher { get; }
-
 #nullable enable
-        internal string? InitialToken { get; }
+        /// <summary>The initial token.</summary>
+        public string? InitialToken { get; set; }
 #nullable restore
+        /// <summary>The token refresher.</summary>
+        public Func<CancellationToken, ValueTask<string>> AsyncTokenRefresher { get; set; }
```

Comment:
This property needs to always be non-null. So we probably need to back it with field to make sure `get` always return a non-null method. Like: https://github.com/Azure/azure-sdk-for-net/pull/18966/files#diff-7c495cc7e44c4a9b8652b611f04e8bfa6d4848c04ff91f7c2941f4c3259bbcecR38

Something like this-ish:
```csharp
private Func<CancellationToken, ValueTask<string>> _asyncTokenRefresher;
public Func<CancellationToken, ValueTask<string>> AsyncTokenRefresher
{
    get => _asyncTokenRefresher ?? _asyncTokenRefresher = (cancellationToken => new ValueTask<string>(tokenRefresher(cancellationToken)));
    set => _asyncTokenRefresher = value;
}
```

Reply:
@RezaJooyandeh I dont think this will work because it requires the `tokenRefresher` which is passed in to the constructor of CommunicationTokenRefreshOptions.cs. I tried to do this null check inside the constructor since devs should have set the asyncTokenRefresher anyway before calling the constructor. https://github.com/Azure/azure-sdk-for-net/pull/18966#discussion_r581462059